telegram/_stars.py

```diff
@@ -237,8 +237,8 @@ def de_json(cls, data: Optional[JSONDict], bot: "Bot") -> Optional["TransactionP
             cls.OTHER: TransactionPartnerOther,
         }
 
-        if cls is TransactionPartner and data.get("type") in _class_mapping:
-            return _class_mapping[data.pop("type")].de_json(data=data, bot=bot)
+        if cls is TransactionPartner and data.get("type") in _class_mapping:  # type: ignore
```

Comment:
had to use the type ignore here, since I'm not doing a:
```python
if not data:
    return None
```
since this breaks the `de_json` for `TransactionPartnerOther` (empty class with only `type` key which is popped so `data` becomes empty and thus `None` is returned). But we used the same approach for `MenuButtonCommands` which works just fine so i'm not sure what I did wrong.

Reply:
ahh such a small thing which I just didn't notice